server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimiter.scala

```diff
@@ -37,7 +39,9 @@ class RedisRateLimiterFactory(redisConfiguration: RedisRateLimiterConfiguration)
     RedisRateLimiter(requestRateLimiter)
   }
 
-  private def convert(rule: Rule): RequestLimitRule = RequestLimitRule.of(rule.duration, rule.quantity.value)
+  private def convert(rule: Rule): RequestLimitRule =
+    RequestLimitRule.of(rule.duration, rule.quantity.value)
+      .withPrecision(Duration.ofSeconds(1))
```

Comment:
by default `precision`  is equal to `duration` 
When we decrease the `precision` value, we will have an algorithm with higher accuracy. 
I'm not sure about the cost of that

Reply:
Maybe you should make this as a parameter no? I would think in test we need precision as it's just gonna be rules of a few seconds max, while on a normal setup we can be a bit more relax as the duration of the limits will be higher and it's less important to be precise?

No idea on the potential cost though. For this we should set up gatling perf tests I guess with different precision if we want to really know

Reply:
> When we decrease the precision value, we will have an algorithm with higher accuracy.

:+1:  higher accuracy will enable more stable tests for sure. We should adopt an accuracy of `1s` in tests.

> I'm not sure about the cost of that

More storage, more complex algorithm. We likely should keep RateLimitJ defauls out of tests.